test/unit/dirty-cache-check.cmake

```
@@ -0,0 +1,58 @@
+include(${CPM_PATH}/CPM.cmake)
+include(${CPM_PATH}/testing.cmake)
+
+set(baseDir "${CMAKE_CURRENT_BINARY_DIR}/test_dirty_cache")
+set(childDir "${baseDir}/edgecase")
+
+find_package(Git REQUIRED)
+
+function(git_do dir)
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} ${ARGN}
```

Also note that the unit tests are failing due to git not being fully configured on CI. I think this could be easily fixed by including them in the git call:

```suggestion
    COMMAND ${GIT_EXECUTABLE}  -c user.name='User' -c user.email='user@email.org' ${ARGN}
```